src/utils/storage.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import FullnodeConnection from '../new/connection';
+import { IStorage, IAddressInfo, IHistoryTx } from '../types';
+import walletApi from '../api/wallet';
+
+import { chunk } from 'lodash';
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import helpers from '../utils/helpers';
+import { deriveAddressP2PKH, deriveAddressP2SH } from '../utils/address';
+
+export async function loadAddresses(startIndex: number, count: number, storage: IStorage): Promise<string[]> {
+  const addresses: string[] = [];
+  const stopIndex = startIndex + count;
+  for (let i = startIndex; i < stopIndex; i++) {
+    const storageAddr = await storage.getAddressAtIndex(i);
+    if (storageAddr !== null) {
+      // This address is already generated, we can skip derivation
+      addresses.push(storageAddr.base58);
+      continue;
+    }
+    // derive address at index i
+    let address: IAddressInfo;
+    if (await storage.getWalletType() === 'p2pkh') {
+      address = await deriveAddressP2PKH(i, storage);
+    } else {
+      address = await deriveAddressP2SH(i, storage);
+    }
+    await storage.saveAddress(address);
+    addresses.push(address.base58);
+  }
+
+  return addresses;
+}
+
+export async function reloadStorage(storage: IStorage, connection: FullnodeConnection) {
+  // unsub all addresses
+  for await (const address of storage.getAllAddresses()) {
+    connection.unsubscribeAddress(address.base58);
+  }
+  const accessData = await storage.getAccessData();
+  if (accessData != null) {
+    // Clean entire storage
+    await storage.cleanStorage(true, true);
+    // Reset access data
+    await storage.saveAccessData(accessData);
+  }
+  return syncHistory(0, await storage.getGapLimit(), storage, connection);
+}
+
+export async function syncHistory(startIndex: number, count: number, storage: IStorage, connection: FullnodeConnection, processHistory: boolean = false) {
```

Comment:
Not really, the pre-calculated addresses are saved to storage and here we use `getAllAddresses` which iterate on these saved addresses to load the history.
On the `loadAddresses` function above we can see that it only derives the address if it not on storage, this is because we should have already pre-calculated and saved the addresses to the storage if we wish to use a list of hardcoded addresses.

Reply:
We can see where the addresses are saved on storage on the [facade's start method](https://github.com/HathorNetwork/hathor-wallet-lib/blob/45ec00a0aadde8de6ade72d4909c44a39e32b0c5/src/new/wallet.js#L1212-L1218)